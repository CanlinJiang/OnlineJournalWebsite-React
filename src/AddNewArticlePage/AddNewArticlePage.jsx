import React from 'react'
import {useState} from 'react'
import {useHistory} from "react-router-dom"
import postData from "../UtilitiesFunctions/postData"
import styles from "./AddNewArticlePage.module.css"
import Frame from "../UtilitiesComponents/MainFrame/Frame"

export default function AddNewArticlePage() {

const [title, setTitle] = useState(null)
const [mood, setMood] = useState(null)
const [status, setStatus] = useState(null)
const [body, setBody] = useState(null)
const [shortQuote, setShortQuote] = useState(null)
const [author, setAuthor] = useState(null)
const [date, setDate] = useState(null)
const [length, setLength] = useState(null)
const [imageUrl, setImageUrl] = useState(null)
const [originalUrl, setOriginalUrl] = useState(null)
const histroy = useHistory();



const handleSubmit = (e) => {
    e.preventDefault();
    const article = {title, mood, status, body, shortQuote, author, date, length, image_url: imageUrl, original_url: originalUrl};
    const url = global.url;
    postData({url, article, actionAfterPost})
    
}

const actionAfterPost = () => {
    alert("Update successfully!")
    histroy.push("/randomwalk");
}

const BodyComponent = () => (
    <div>
            <form className = {styles.flexContainer} id="newArticle" onSubmit = {handleSubmit} >

                <div className={styles.topContainer}>
                <input className={styles.title} type="text" placeholder="Title:" required onChange={(e) => setTitle(e.target.value)}></input>
                <select className={styles.topButton} onChange = {(e) => setMood(e.target.value)}>
                    <option disabled selected>Mood</option>
                    <option value="Joyful">Joyful</option>
                    <option value="Powerful">Powerful</option>
                    <option value="Peaceful">Peaceful</option>
                    <option value="Sad">Sad</option>
                    <option value="Mad">Mad</option>
                    <option value="Scared">Scared</option>
                </select>
                <select className={styles.topButton} onChange = {(e) => setStatus(e.target.value)}>
                    <option disabled selected>Read/unread</option>
                    <option value="Read">Read</option>
                    <option value="Unread">Unread</option>
                </select>
                </div>

                
                <textarea className={styles.bodyContainer} form="newArticle" onChange = {(e) => setBody(e.target.value)}>Enter article here...</textarea>



                <div className={styles.bottomContainer}>

                    <textarea className={styles.shortQuote} form="newArticle" onChange = {(e) => setShortQuote(e.target.value)}>Enter short quote here...</textarea>
                    
                    <div className={styles.columnLinkContainer}>
                    <form className={styles.length} onChange = {(e) => setLength(e.target.value)}>
                        <div><input type="radio" id="short" name="length" value="Short"></input><label for = "short"> Short</label></div>
                        <div><input type="radio" id="medium" name="length" value="Medium"></input><label for = "medium"> Medium</label> </div>   
                        <div><input type="radio" id="long" name="length" value="Long"></input><label for = "long"> Long</label></div>
                    </form>
                    <input className={styles.link} type="text" placeholder="Image Link: " pattern="https?://.+" onChange={(e) => setImageUrl(e.target.value)}></input>
                    <input className={styles.link} type="text" placeholder="Original Article Link:" pattern="https?://.+" onChange={(e) => setOriginalUrl(e.target.value)}></input>
                    </div>
                </div>

                <div class={styles.lastRowContainer}>
                        <input type="text" placeholder="Author:" required onChange={(e) => setAuthor(e.target.value) }></input>
                        <input type="text" placeholder="Date: 2021-04-21" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" required onChange={(e) => setDate(e.target.value)}></input>
                        <input type="submit"></input>
                    </div>

                
                


                

            </form>

            
        </div>
)



    return (
        <Frame BodyComponent={BodyComponent}/>
    )
}
