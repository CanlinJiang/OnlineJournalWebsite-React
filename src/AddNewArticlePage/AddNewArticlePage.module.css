.topContainer{
    width: 80%;
    display: flex;
    flex-flow: row wrap;
    margin-bottom: 25px;
}

.title{
    flex-grow: 9;
    margin-right: 20px;
    padding:10px;
    border: 2px solid black;
    border-radius: 5px;
}

.topButton{
    flex-grow: 1;
    margin-right: 20px;
    padding:10px;
    border: 2px solid black;
    border-radius: 5px;
}

.topButton:last-child{
    flex-grow: 1;
    margin-right: 0px;
    padding:10px;
    border: 2px solid black;
    border-radius: 5px;
}

.bodyContainer{
    display: flex;
    width: 80%;
    height: 400px;
    align-self: center;
    margin-bottom: 25px;
    padding:30px;
    border: 2px solid black;
    border-radius: 5px;
}

.bottomContainer{
    width: 80%;
    display: flex;
    padding-right:10px;
}

.shortQuote{
    flex-grow: 5;
    padding:30px;
    border: 2px solid black;
    border-radius: 5px;
}

.columnLinkContainer {
    flex-grow: 1;
    display: flex;
    flex-flow: column wrap;
    margin:0;
}
.columnLinkContainer input[type="radio"]{
    padding:10px;
    margin-top: 10px;
    margin-left: 10px;
    border: 2px solid black;
    border-radius: 5px;
}
.link {
    width: 100%;
    padding:10px;
    margin-top: 10px;
    margin-left: 10px;
    border: 2px solid black;
    border-radius: 5px;
    
}
.length {
    padding:10px;
    margin-left: 10px;
    border: 2px solid black;
    border-radius: 5px;
    display: flex;
    flex-wrap: nowrap;
    justify-content: space-evenly;
    align-items:center;
}

.lastRowContainer{
    margin-top: 25px;
    width: 80%;
    display: flex;
    flex-flow: row wrap;
    padding-left: 0px;
}

.lastRowContainer input{
    padding:10px;
    margin-top: 0px;
    margin-right: 10px;
    border: 2px solid black;
    border-radius: 5px;
}
.lastRowContainer input[type="submit"]{
    padding:10px 30px;
    border: 2px solid black;
    border-radius: 5px;
    transition: color 1s, background-color 0.5s;
}

.lastRowContainer input[type="submit"]:hover{
    padding:10px 30px;
    border: 2px solid black;
    border-radius: 5px;
    background-color: black;
    color:white;
}

.flexContainer{
    display: flex;
    flex-flow: column wrap;
    width: 100%;
    align-items: center;
}

@media (max-width: 768px) {
    .flexContainer, .topContainer, .bodyContainer, .bottomContainer, .columnLinkContainer, .lastRowContainer, .length, .link, .shortQuote, .title, .topButton{
        display: block;
        width: 100%;
        margin-top: 10px;
    }
    .shortQuote{
        margin-left: 10px;
    }
    .bottomContainer{
        margin-left: -10px;
        width: 103%;
    }

    .lastRowContainer input{
        margin-bottom: 10px;
        width: 100%;
    }
}