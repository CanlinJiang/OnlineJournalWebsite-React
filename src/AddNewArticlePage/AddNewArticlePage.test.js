import AddNewArticlePage from "./AddNewArticlePage"
import React from "react";

describe("AddNewArticlePage tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<AddNewArticlePage />);
      expect(container).toMatchSnapshot();
    });
  
  });