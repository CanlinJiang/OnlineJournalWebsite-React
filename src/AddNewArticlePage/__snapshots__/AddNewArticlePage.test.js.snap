// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewArticlePage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="navigation"
      >
        <ul
          class="leftBar"
        >
          <li>
            <a
              href="/minelist"
            >
              Mine's
            </a>
          </li>
          <li>
            <a
              href="/moodchoosing"
            >
              Moods
            </a>
          </li>
          <li>
            <a
              href="/randomwalk"
            >
              Random Walk
            </a>
          </li>
        </ul>
        <ul
          class="rightBar"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <div>
        <form
          class="flexContainer"
          id="newArticle"
        >
          <div
            class="topContainer"
          >
            <input
              class="title"
              placeholder="Title:"
              required=""
              type="text"
            />
            <select
              class="topButton"
            >
              <option
                disabled=""
              >
                Mood
              </option>
              <option
                value="Joyful"
              >
                Joyful
              </option>
              <option
                value="Powerful"
              >
                Powerful
              </option>
              <option
                value="Peaceful"
              >
                Peaceful
              </option>
              <option
                value="Sad"
              >
                Sad
              </option>
              <option
                value="Mad"
              >
                Mad
              </option>
              <option
                value="Scared"
              >
                Scared
              </option>
            </select>
            <select
              class="topButton"
            >
              <option
                disabled=""
              >
                Read/unread
              </option>
              <option
                value="Read"
              >
                Read
              </option>
              <option
                value="Unread"
              >
                Unread
              </option>
            </select>
          </div>
          <textarea
            class="bodyContainer"
            form="newArticle"
          >
            Enter article here...
          </textarea>
          <div
            class="bottomContainer"
          >
            <textarea
              class="shortQuote"
              form="newArticle"
            >
              Enter short quote here...
            </textarea>
            <div
              class="columnLinkContainer"
            >
              <form
                class="length"
              >
                <div>
                  <input
                    id="short"
                    name="length"
                    type="radio"
                    value="Short"
                  />
                  <label
                    for="short"
                  >
                     Short
                  </label>
                </div>
                <div>
                  <input
                    id="medium"
                    name="length"
                    type="radio"
                    value="Medium"
                  />
                  <label
                    for="medium"
                  >
                     Medium
                  </label>
                   
                </div>
                <div>
                  <input
                    id="long"
                    name="length"
                    type="radio"
                    value="Long"
                  />
                  <label
                    for="long"
                  >
                     Long
                  </label>
                </div>
              </form>
              <input
                class="link"
                pattern="https?://.+"
                placeholder="Image Link: "
                type="text"
              />
              <input
                class="link"
                pattern="https?://.+"
                placeholder="Original Article Link:"
                type="text"
              />
            </div>
          </div>
          <div
            class="lastRowContainer"
          >
            <input
              placeholder="Author:"
              required=""
              type="text"
            />
            <input
              pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
              placeholder="Date: 2021-04-21"
              required=""
              type="text"
            />
            <input
              type="submit"
            />
          </div>
        </form>
      </div>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
